### DISCUSSION TITLE
Wiring PiZero2w with ItsyBitsyRP2040

### DISCUSSION THREAD

**Discussion by sysTim1977** [awawa-dev/HyperSPI#48] [Q&A] [answered]
1st. Thanks for this great piece of software. 

With my new TV I want to use new LED (Sk6812) with an Adafruit "ItsyBitsy RP2040" LED Driver.
I use a RaspberryPizero2w (NOT zero, shown in the picture) and the second USB-Port is used by Grabber, so I have to use SPI.

Can anybody have a look on my wiring? I want to ask before I break something.

I don't want to use a second USB-Power, so I want to use the LED-Power-Supply for the LED-Driver.

-=BEGIN Wiring  Pizero2w - Itsy=-
Power-Wiring: 
Powersupply (5v) beginning and end of the stripe and BAT of the "Itsy", GND to GND of the stripe (both ends) the Itsy and the PiZero2w (black and red wire in the picture).

Data-Wiring: 
Itsy D5/GP14 (level shifted Output) -> LED-Din (yellow wire in picture)

SPI-Wiring:
PiZero2w GPIO12, MOSI (Pin19) -> Itsy MISO (Cyan Wire in Picture)
PiZero2w GPIO13, MISO (Pin21) -> Itsy MOSI (White Wire in Picture)
PiZero2w GPIO14, SCLK (Pin23) -> Itsy SCK (Violet Wire in Picture)
PiZero2w GPIO10, CE0 (Pin24) -> Itsy D3/GP05, SPI0 CS (Orange Wire in Picture)
-=END Wiring Setup RP2Wzero - Itsy=-

Did anybody see any Errors in this setup?
Is HyperSPi the right driver or have I use the HyperSerialPico?

Thanks for your answers.

![PiZero2W-ItsyBitsyRP2040](https://github.com/awawa-dev/HyperSPI/assets/163929358/0b742b97-c0e9-4f48-be3f-955b6cfd8801)

**Comment by awawa-dev** [comment]
Hi
I only tested generic rp2040 with HyperSPI. If you don't get a response from other users, I will check it with ItsyBitsyRP2040
 and prepare the firmware during this weekend.

**Reply by sysTim1977** [reply]
Thanks for this fast answer.
The generall wiring seems ok?

**Reply by awawa-dev** [reply]
It's too late at night for me to check it ;) As long as you don't make a mistake and supply 5V or higher voltageto some GPIO with a 3.3V input, nothing should break.

**Comment by awawa-dev** [comment]
I created PR to support ItsyBitsyRP2040 board directly https://github.com/awawa-dev/HyperSPI/pull/49
Build: https://github.com/awawa-dev/HyperSPI/actions/runs/9620717202 or you can fork & trigger the build yourself
in the artifacts.zip there is Adafruit_ItsyBitsy_RP2040.zip which contains various firmware for this board.

I decided to use a different set of pins ( ~~GPIO27~~ GPIO28, GPIO26, GPIO29) on the other side of the 5V output because a mistake with it would be disastrous when connected to the RPi. Beside I don't have any soldered pins on the top of the board and it would be a very bad solution for prototype boards. I haven't tested it yet.

**Comment by awawa-dev** [comment]
I made one mistake with GPIO27 (it should be GPIO28) but I corrected it in the pull request and already tested new firmware. Works fine:
![obraz](https://github.com/awawa-dev/HyperSPI/assets/69086569/483b9981-70ae-496c-8b85-501459e77655)
I used my standard cable connector that I used for most of others HyperSPI projects.

**Comment by sysTim1977** [comment, marked as answer]
Thanks, it works very fine.
For all the other "noobs" like me, here is the wiring.

![PiZero2w-ItsyBitsySPI](https://github.com/user-attachments/assets/e4e9d1c7-a76f-447f-9d17-ca7c908567c6)
